app/main.py

```diff
@@ -1,8 +1,24 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_people = []
+    for person_dict in people:
+        person = Person(name=person_dict["name"], age=person_dict["age"])
+        person_people.append(person)
+
+    for person_dict in people:
+        person = Person.people[person_dict["name"]]
```

Comment:
Accessing `Person.people` by name assumes that all names are unique, which might not be the case. This could lead to incorrect spouse assignments if names are not unique.